### PR TITLE
refactor: (ship ahead 2) a couple feature tweaks and several shared components

### PR DESCRIPTION
**Pull request by uchilaka**
## Jira Card

[STORYAI-###](https://larcity.atlassian.net/browse/STORYAI-###)

## Summary

> Focus on what's important to note here. The Jira card should provide the rest the context.

## Review app(s)

- <storyteller-ui-git-###-donnadieu.vercel.app>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the README/confluence documentation.
- [ ] I have updated the relevant documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/features/GoogleLogin/index.tsx
Comment (uchilaka) [inline src/features/GoogleLogin/index.tsx:1]: Implements `features/AppProgress` so we can show a loading skeleton or spinner when app-interrupting stuff is happening like logging in

File: src/components/shared/ScriptTag.tsx
Comment (uchilaka) [inline src/components/shared/ScriptTag.tsx:1]: This is replaced by `features/ScriptTag`

File: src/features/StoryBuilder/Status.tsx
Comment (uchilaka) [inline src/features/StoryBuilder/Status.tsx:1]: Extracted re-usable components into the new shared components:
- `ResponsiveTextBox.tsx`
- `TitleTextBox.tsx`

File: src/components/shared/AuthProvider.tsx [hunk 1]
```diff
@@ -43,20 +46,27 @@ export function AuthStatus() {
...
+    </Box>
   )
 }
 
 export function RequireAuth({ children }: WithRequiredChildren) {
   let auth = useAuth()
   let location = useLocation()
+  const [params] = useSearchParams()
+
+  const loginPath = params.size > 0 ? `/login?${params.toString()}` : '/login'
```
Comment (uchilaka) [inline src/components/shared/AuthProvider.tsx:58]: Now supports GET parameters if we detect them, like `u=User;8` which would be how we could see stuff feature flagged for `User.find(8)` without being logged in (I intend to use this later to test Apple Login without turning it on for everyone else)

File: src/components/Login.tsx
Comment (uchilaka) [inline src/components/Login.tsx:1]: Implements:
- `features/AppProgress`
- `LoadingAnimation.tsx`

File: src/App.tsx [hunk 2]
```diff
@@ -26,42 +27,44 @@ function App() {
...
+                      />
+                      <Route index element={<Navigate to="stories/new" />} />
+                    </Route>
+                  </Route>
+                  <Route path="/v0">
                     <Route
                       path="stories/:action"
                       element={
                         <RequireAuth>
-                          <StoryGeneratorV1 />
```
Comment (Donnadieu) [inline src/App.tsx:40]: Why are we removing the version?

**Review by Donnadieu — APPROVED** [review]

File: src/App.tsx (hunk 2 above)
Comment (uchilaka) [reply src/App.tsx:40]: We didn't - just moved the route for `V1` above that for `V0`

**Pull request merged**